Managed Test Port

bryanar opened the pull request:
This is a collection of managed runtime tests from an internal legacy test tree.

akoeplinger commented:
There are a lot of merge conflicts in the files:

```
$ grep -r "<<<<<<<" .
./tests/src/Common/CoreCLRTestLibrary/project.json.orig:<<<<<<< HEAD
./tests/src/Common/CoreCLRTestLibrary/project.json.orig:<<<<<<< HEAD
./tests/src/Common/Coreclr.TestWrapper/project.json.orig:<<<<<<< HEAD
./tests/src/Common/Coreclr.TestWrapper/project.json.orig:<<<<<<< HEAD
./tests/src/Threading/ThreadStatics/project.json.orig:<<<<<<< HEAD
./tests/src/Threading/ThreadStatics/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/NativeLibs/project.lock.json.orig:<<<<<<< HEAD
./tests/src/Loader/typeinitialization/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/typeinitialization/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/polyrec/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/polyrec/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/classloader-generics/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/classloader-generics/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/classloader/project.json.orig:<<<<<<< HEAD
./tests/src/Loader/regressions/classloader/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/SIMD/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/SIMD/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/minimal/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/minimal/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/threading/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/threading/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/empty/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/empty/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/threading+thread/project.json.orig:<<<<<<< HEAD
./tests/src/JIT/config/threading+thread/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Stress/Tests/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Stress/Tests/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Stress/Framework/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Stress/Framework/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/Boxing/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/DoublinkList/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Scenarios/FinalNStruct/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/Pinning/PinningOther/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Features/HeapExpansion/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/485617/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/476725/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/426480/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta2/471729/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/494226/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-rtm/544701/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v3.0-mq/573277/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/289745/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Regressions/v2.0-beta1/149926/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json.orig:<<<<<<< HEAD
./tests/src/GC/Coverage/project.json.orig:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json.orig:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json.orig:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json.orig:<<<<<<< HEAD
./tests/src/GC/API/GCSettings/project.json.orig:<<<<<<< HEAD
./tests/src/dir.targets.orig:<<<<<<< HEAD
./tests/src/Interop/NativeCallable/project.json.orig:<<<<<<< HEAD
./tests/src/Interop/NativeCallable/project.json.orig:<<<<<<< HEAD
./tests/src/Interop/ReversePInvoke/Marshalling/project.json.orig:<<<<<<< HEAD
./tests/src/Interop/ReversePInvoke/Marshalling/project.json.orig:<<<<<<< HEAD
./tests/src/Interop/ICastable/project.json.orig:<<<<<<< HEAD
./tests/src/Interop/ICastable/project.json.orig:<<<<<<< HEAD
./tests/src/Exceptions/Finalization/project.json.orig:<<<<<<< HEAD
./tests/src/Exceptions/Finalization/project.json.orig:<<<<<<< HEAD
./tests/src/CLRTest.Execute.Batch.targets.orig:<<<<<<< HEAD
./tests/src/runtime/project.json.orig:<<<<<<< HEAD
./tests/src/runtime/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/assemblyref/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/assemblyref/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/common/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/common/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/expl_double/project.json.orig:<<<<<<< HEAD
./tests/src/Regressions/expl_double/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/testv1/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/testv1/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/testv2/project.json.orig:<<<<<<< HEAD
./tests/src/readytorun/testv2/project.json.orig:<<<<<<< HEAD
./tests/src/CLRTest.Execute.Bash.targets.orig:<<<<<<< HEAD
./tests/src/managed/Compilation/project.json.orig:<<<<<<< HEAD
./tests/src/managed/Compilation/project.json.orig:<<<<<<< HEAD
./tests/dir.props.orig:<<<<<<< HEAD
./src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs.orig:<<<<<<< HEAD
```

bryanar commented:
Whoah, there certainly are. I am on it.

bryanar commented:
Thanks for catching that @akoeplinger! I've amended the commit.

jkotas commented:
There is a ton of tests that are duplicated in CoreFX and/or that should not be in CoreCLR at all - because of they are testing stuff high-level libraries implemented in CoreFX.


Example of duplicate test between CoreCLR and CoreFX:

https://github.com/bryanar/coreclr/blob/6fb81eda453c8944d026256aea15bd90e83d5fc6/tests/src/CoreMangLib/system/collections/generic/collectiontesting/genericicollectiontest.cs#L56
vs.
https://github.com/dotnet/corefx/blob/master/src/System.Collections/tests/Generic/Common/GenericICollectionTest.cs#L57


Example of tests that are testing high-level service (System.Diagnostic.Process):

https://github.com/bryanar/coreclr/tree/6fb81eda453c8944d026256aea15bd90e83d5fc6/tests/src/CoreMangLib/components/process

Notice that majority of the System.Diagnostic.Process tests are very Windows specific - will require rewrite for cross-plat. This rewrite seems to have been done in CoreFx already.


Can we prune this commit to eliminate these categories?

cc @weshaggard

jkotas commented:
@weshaggard do we happen to have a log somewhere of which internal tests have been ported to corefx?

jkotas commented on src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs.orig:
```diff
@@ -0,0 +1,99 @@
+            }
+
+            string result = string.FastAllocateString(s.Length);
+<<<<<<< HEAD
```
Conflict

bryanar replied:
Deleted. I wasn't aware of the difftool leaving *.orig files everywhere. Did a scan and removed them all as well as any other merge remnants.

jkotas commented:
Also, this should be submitted as dotnet-bot

bryanar commented:
Hey Jan, when it came to removing duplicates, I wasn't entirely rigorous so there is likely to be some remnants hanging around (i.e. my reasoning was along the lines of 'avoid directory named corefx in legacy tree). What I think I can do is grab the corefx repo and diff the *.cs files in that against these guys and see which are being duplicated (i.e. detect which files have a lot in common and remove those specific test cases). 

One other question, what is dotnet-bot? How do I use it? etc.

jkotas commented:
> What I think I can do is grab the corefx repo and diff the *.cs files in that against these guys and see which are being duplicated 

Sounds good.


> One other question, what is dotnet-bot? How do I use it?

Commit it as `--author=”dotnet-bot <dotnet-bot@microsoft.com>”`. We use dotnet-bot account as author for the initial commits of large amounts existing code so that it is clear that it was actually written by whoever is doing the commit.

bryanar commented:
I've amended the commit author, and I have a script running over the source files right now looking for duplicates. Tomorrow I'll interpret the results and make the appropriate cleanups.

bryanar commented:
Woooo, that took longer than expected. I've removed all source files that were overlapping. Additionally, I found I had a ton of directories that contained orphaned source code, as well some directories that just contained app.config/project.json files. I've removed these and so the commit is a bit lighter/cleaner now.

akoeplinger commented:
The file `tests/errors.txt`seems unnecessary as well.

bryanar commented:
@akoeplinger agreed. removed.

weshaggard commented:
@jkotas we don't have a log of the tests that were ported but I agree we should eliminate the duplication. @bryanar how confident are you of your process in eliminating duplication?

jkotas commented:
I still see a lot of duplication, in particular under CoreMangLib. 

One example from many:

https://github.com/dotnet/coreclr/blob/master/tests/src/CoreMangLib/cti/system/boolean/booleancompareto_boolean.cs

https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/Boolean.cs#L176

bryanar commented:
Ah, I took the word "duplication" a little more loosely. My diffing tool explicitly just checked for file similarity of the files, not duplication of scenario coverage. I'm not sure I see a problem with the duplication of scenarios in this way though. 

The way I am looking at it is that these tests are here to act as a first line of defense against code that may break the runtime. I am not sure how often we're testing the latest runtimes against the corefx libraries, but if it isn't immediately, then I'd argue that having at least some test cases earlier on in the process is better. Since they don't run by default, I don't see them as a hindrance to developer time either.
